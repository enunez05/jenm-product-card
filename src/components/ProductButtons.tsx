import React, { useCallback, useContext } from 'react';
import '../styles/styles.css';
import { ProductContext } from './ProductCard';

export interface Props {
	className?:string;
	style?: React.CSSProperties
}

export const ProductButtons = ({className}:Props) => {
	const { counter, increaseBy, maxCount } = useContext(ProductContext);


	const isMyReached = useCallback(
	  () => !!maxCount && counter === maxCount,
	  [counter, maxCount],
	)
	
	return (
		<div className={`buttonsContainer ${className}`}>
			<button className={`buttonMinus`} onClick={()=> increaseBy(-1)}> - </button>
			<div className={`countLabel`}>{counter}</div>
			<button className={`buttonAdd ${ isMyReached() && 'disabled'}`} onClick={()=> increaseBy(1)}>+</button>
		</div>
	)
}