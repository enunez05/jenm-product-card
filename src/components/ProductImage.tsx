import React, { useContext } from "react";
import { ProductContext } from "./ProductCard";
import '../styles/styles.css';

export interface Props {
	img?:string,
	className?:string,
	style?: React.CSSProperties
}

export const ProductImage = ({img, className, style}:Props) => {

	const {product} = useContext(ProductContext);
	let imgToShow:string;

	if(img){
		imgToShow= img;
	} else if (product.img) {
		imgToShow= product.img
	} else {
		imgToShow= 'https://picsum.photos/200/300';
	}

	return( <img style={style} className={`productImg ${className}`} src={imgToShow} alt="Coffe Mug" />)
}