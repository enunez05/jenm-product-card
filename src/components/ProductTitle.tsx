import React, { useContext } from "react"
import { ProductContext } from "./ProductCard";
import '../styles/styles.css';

export interface Props {
	title?:string,
	className?:string
	activeClassName?:string,
	style?: React.CSSProperties
}

export const ProductTitle = ({title, className, activeClassName}:Props) => {
	const {product} = useContext(ProductContext)
	return (
		<span className={`productDescription ${className} ${activeClassName}`}>
			{title ? title : product.title}
		</span>
	)
}