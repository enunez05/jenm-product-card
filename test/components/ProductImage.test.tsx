import * as React from 'react';
import renderer from 'react-test-renderer'
import  {ProductCard, ProductImage } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () =>{
    test(('Debe de mostrar el componente con el src del producto'), () => {
        const wrapper = renderer.create(
            <ProductImage img={product2.img} />
        )
        expect(wrapper.toJSON()).toMatchSnapshot()
    });
    test('Debe de mostrar el componente padre con el componente imagen con la imagen',() => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage img={product2.img} />
                    )
                }
            </ProductCard>
        )
        expect(wrapper.toJSON()).toMatchSnapshot()
    });
});
