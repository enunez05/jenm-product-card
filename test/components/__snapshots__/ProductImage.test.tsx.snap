// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe de mostrar el componente con el src del producto 1`] = `
<img
  alt="Coffe Mug"
  className="productImg undefined"
  src="./coffee-mug.png"
/>
`;

exports[`ProductImage Debe de mostrar el componente padre con el componente imagen con la imagen 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Coffe Mug"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
